internal/blockchain/ethereum/private_transaction_manager.go

```diff
@@ -0,0 +1,262 @@
+
+type PrivateKeyData struct {
+	Bytes string `json:"bytes"`
+}
+
+type PrivateKey struct {
+	Type string         `json:"type"`
+	Data PrivateKeyData `json:"data"`
+}
+
+func CreateTesseraKeys(ctx context.Context, image, outputDirectory, prefix, name, password string) (privateKey, pubKey, path string, err error) {
+	// generates both .pub and .key files used by Tessera
+	var filename string
+	if prefix != "" {
+		filename = fmt.Sprintf("%v_%s", prefix, name)
+	} else {
+		filename = name
+	}
+	if err := os.MkdirAll(outputDirectory, 0755); err != nil {
+		return "", "", "", err
+	}
+	fmt.Println("generating tessera keys")
+	err = docker.RunDockerCommand(ctx, outputDirectory, "run", "--rm", "-v", fmt.Sprintf("%s:/keystore", outputDirectory), image, "-keygen", "-filename", fmt.Sprintf("/keystore/%s", filename))
+	if err != nil {
+		return "", "", "", err
+	}
+	path = fmt.Sprintf("%s/%s", outputDirectory, filename)
+	pubKeyBytes, err := os.ReadFile(fmt.Sprintf("%v.%s", path, "pub"))
+	if err != nil {
+		return "", "", "", err
+	}
+	privateKeyBytes, err := os.ReadFile(fmt.Sprintf("%v.%s", path, "key"))
+	if err != nil {
+		return "", "", "", err
+	}
+	var privateKeyData PrivateKey
+	err = json.Unmarshal(privateKeyBytes, &privateKeyData)
+	if err != nil {
+		return "", "", "", err
+	}
+	return privateKeyData.Data.Bytes, string(pubKeyBytes[:]), path, nil
+}
+
+func CopyTesseraKeysToVolume(ctx context.Context, tesseraKeyPath, volumeName string) error {
+	if err := docker.MkdirInVolume(ctx, volumeName, tmpath); err != nil {
+		return err
+	}
+	if err := docker.CopyFileToVolume(ctx, volumeName, filepath.Join(tesseraKeyPath, "tm.pub"), tmpath); err != nil {
+		return err
+	}
+	if err := docker.CopyFileToVolume(ctx, volumeName, filepath.Join(tesseraKeyPath, "tm.key"), tmpath); err != nil {
+		return err
+	}
+	return nil
+}
+
+func CreateTesseraEntrypoint(ctx context.Context, outputDirectory, volumeName, memberCount string) error {
+	// only tessera v09 onwards is supported
+	var sb strings.Builder
+	memberCountInt, _ := strconv.Atoi(memberCount)
+	for i := 0; i < memberCountInt; i++ {
+		sb.WriteString(fmt.Sprintf("{\"url\":\"http://member%dtessera:9000\"},", i)) // construct peer list
+	}
+	peerList := strings.TrimSuffix(sb.String(), ",")
+	content := fmt.Sprintf(`export JAVA_OPTS="-Xms128M -Xmx128M"
+DDIR=/data/qdata/dd
+mkdir -p ${DDIR}
+[ -f ${DDIR}/dd/tm.ipc ] && rm -f ${DDIR}/dd/tm.ipc
+cat <<EOF > ${DDIR}/tessera-config-09.json
+	{
+		"useWhiteList": false,
+		"jdbc": {
+			"username": "sa",
+			"password": "",
+			"url": "jdbc:h2:./${DDIR}/db;TRACE_LEVEL_SYSTEM_OUT=0",
+			"autoCreateTables": true
+		},
+		"serverConfigs":[
+			{
+				"app":"ThirdParty",
+				"enabled": true,
+				"serverAddress": "http://$(hostname -i):%s",
+				"communicationType" : "REST"
+			},
+			{
+				"app":"Q2T",
+				"enabled": true,
+				"serverAddress": "http://$(hostname -i):%s",
+				"sslConfig": {
+					"tls": "OFF"
+				},
+				"communicationType" : "REST"
```

Comment:
If the Quorum to Tessera comms is configured via REST, why do we also have UNIX domain socket configuration above?

Reply:
Will remove the above check and removal of the Unix domain socket configuration. We intentionally want to set q2t to REST since we made the choice to not share storage between Quorum and Tessera.